aws-xray-recorder-sdk-sql/src/test/java/com/amazonaws/xray/sql/SqlSubsegmentsTest.java

```diff
@@ -102,4 +102,15 @@ void testCreateSubsegmentWithSql() throws SQLException {
         assertThat(sub.getNamespace()).isEqualTo(Namespace.REMOTE.toString());
         assertThat(sub.getSql()).containsAllEntriesOf(expectedSqlParams);
     }
+
+    @Test
+    void testCreateSubsegmentWhenConnectionThrowsException() throws SQLException {
+        when(connection.getMetaData()).thenThrow(new SQLException());
+
+        Subsegment sub = SqlSubsegments.forQuery(connection, SQL);
+
+        assertThat(AWSXRay.getCurrentSubsegment()).isEqualTo(sub);
+        assertThat(sub.isInProgress()).isTrue();
+        assertThat(sub.getParentSegment().getSubsegments()).containsOnly(sub);
```

Comment:
I think the intent of this check is captured better as just `contains`

Reply:
Think we can use an assertion on the subsegment name, both as a check and documentation of what happens in this case.

Reply:
Good calls